docs/basics/accounts.md

```diff
@@ -56,80 +56,97 @@ To derive `PubKey`s and `PrivKey`s, the Cosmos SDK uses a standard called [BIP32
+
+Similarly to `Address`, bech32 is used to format `PubKey` and for all communication with a blockchain:
 
 +++ https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc3/types/address.go#L579-L729
 
-### Addresses
 
-The Cosmos SDK comes by default with 3 types of addresses:
+## Keyring
 
-- `AccAddress` for accounts.
-- `ValAddress` for validator operators.
-- `ConsAddress` for validator nodes.
+A `Keyring` is an object that stores and manages accounts. In the Cosmos SDK, a `Keyring` implementation follows the `Keyring` interface:
 
-Each of these address types are an alias for an hex-encoded `[]byte` array of length 20. Here is the standard way to obtain an address `aa` from a `Pubkey pub`:
++++ https://github.com/cosmos/cosmos-sdk/blob/v0.42.1/crypto/keyring/keyring.go#L51-L92
 
-```go
-aa := sdk.AccAddress(pub.Address().Bytes())
-```
+The default implementation of `Keyring` comes from the third-party [`99designs/keyring`](https://github.com/99designs/keyring) library.
+
+A few notes on the `Keyring` methods:
+
+- `Sign(uid string, payload []byte) ([]byte, sdkcrypto.PubKey, error)` strictly deals with the signature of the `payload` bytes. Some preliminary work should be done beforehand to prepare and encode the transaction into a canonical `[]byte` form. Protobuf being not deterministic, it has been decided in [ADR-020](../architecture/adr-020-protobuf-transaction-encoding.md) that the canonical `payload` to sign is the `SignDoc` struct, deterministically encoded using [ADR-027](adr-027-deterministic-protobuf-serialization.md). Note that signature verification is not implemented in the SDK by default, it is deferred to the [`anteHandler`](../core/baseapp.md#antehandler).
+  +++ https://github.com/cosmos/cosmos-sdk/blob/v0.42.1/proto/cosmos/tx/v1beta1/tx.proto#L47-L64
 
-These addresses implement the `Address` interface:
+- `NewAccount(uid, mnemonic, bip39Passwd, hdPath string, algo SignatureAlgo) (Info, error)` creates a new account based on the [`bip44 path`](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) and persists it on disk (note that the `PrivKey` is [encrypted with a passphrase before being persisted](https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc3/crypto/armor.go), it is **never stored unencrypted**). In the context of this method, the `account` and `address` parameters refer to the segment of the BIP44 derivation path (e.g. `0`, `1`, `2`, ...) used to derive the `PrivKey` and `PubKey` from the mnemonic (note that given the same mnemonic and `account`, the same `PrivKey` will be generated, and given the same `account` and `address`, the same `PubKey` and `Address` will be generated). Finally, note that the `NewAccount` method derives keys and addresses using the algorithm specified in the last argument `--algo`. The following keys are supported by the keyring:
```

Comment:
```suggestion
- `NewAccount(uid, mnemonic, bip39Passwd, hdPath string, algo SignatureAlgo) (Info, error)` creates a new account based on the [`bip44 path`](https://github.com/bitcoin/bips/blob/master/bip-0044.mediawiki) and persists it on disk. The `PrivKey` is **never stored unencrypted**, instead it is [encrypted with a passphrase before being persisted](https://github.com/cosmos/cosmos-sdk/blob/v0.40.0-rc3/crypto/armor.go). In the context of this method, the `account` and `address` parameters refer to the segment of the BIP44 derivation path (for example, `0`, `1`, `2`, ...) that is used to derive the `PrivKey` and `PubKey` from the mnemonic. Using the same mnemonic and `account`, the same `PrivKey` is generated and given the same `account` and `address`, so the same `PubKey` and `Address` are generated. Finally, the `NewAccount` method derives keys and addresses using the algorithm specified in the last argument `--algo`. The following keys are supported by the keyring:
```

Reply:
I _think_ my suggestions make sense, let me know if I introduced errors, I am keen to understand. @robert-zaremba I am learning so much here!

Reply:
even though we want to use future tense, the best tense is present, again we can turn to the [Google develop documentation style guide](https://developers.google.com/style/tense)

Reply:
In fact the I will need to reword it.  We shouldn't use _account_ in derivation mechanism.